File: Pset6/sentimental-readability/readability.py
from cs50 import get_string, get_float

def count_letters(text):
    letters = 0
    for char in text:
        if char.isalpha():
            letters += 1
    return letters

def count_words(text):
    words = 1
    for char in text:
        if char == ' ':
            words += 1
    return words

def count_sentences(text):
    sentences = 0
    for char in text:
        if char in ['.', '?', '!']:
            sentences += 1
    return sentences

def grade_function(text):
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    index = 0.0588 * (letters * 100 / words) - 0.296 * (sentences * 100 / words) - 15.8
    grade = round(index)

    if index < 1:
        print("Before Grade 1")
        return 0
    elif index >= 16:
        print("Grade 16+")
        return 0
    else:
        print(f"Grade {grade}")

    return index

def main():
    text = input("Text: ")
    grade = grade_function(text)

if __name__ == "__main__":
    main()

File: Pset2/readability/readability.c
//#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
float grade_function(string text);

int main(void)
{

    // TODO
    // ask user for "Text" using string = get_string.
    prinft("Text: ");
    scantf("%s", text);
    float grade = grade_function(text);

    return 0;
} // END OF MAIN

// count letters function
// IF Char is NOT Alphanumeric do not count.
// ELSE IF count char/string.
// other way to do this funtion is with ISALNUM but couldnt figure out how to do it
// used lecture example instead.
int count_letters(string text)
{

    int letters = 0; // intialize int letters to 0

    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
    }

    return letters;
}

int count_words(string text)
{
    int words = 1; // words initialize at 1 for if they start at 0, we get 1 less word than expected.
    for (int i = 0; i < strlen(text); i++)
        if (text[i] == ' ')
        {
            words++;
        }
    return words;
}

int count_sentences(string text)
{

    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
        if ((text[i] == '.') || (text[i] == '?') || (text[i] == '!'))
        {
            sentences++;
        }

    return sentences;
}
float grade_function(string text)
{

    float index =
        0.0588 * count_letters(text) * 100 / count_words(text) - 0.296 * count_sentences(text) * 100 / count_words(text) - 15.8;
    int grade = round(index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    else
    {
        printf("Grade %i\n", grade);
    }

    return index;
}
